File: src/main/java/ee/upcourse/trainingmanager/model/Lesson.java
package ee.upcourse.trainingmanager.model;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import ee.upcourse.trainingmanager.config.SqlTimeDeserializer;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@Entity
@Table(name = "lesson")
public class Lesson {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String subject;

    private LocalDateTime startTime;

    private LocalDateTime endTime;

    @ManyToOne(cascade = {CascadeType.DETACH, CascadeType.MERGE, CascadeType.PERSIST, CascadeType.REFRESH})
    @JoinColumn(name = "topic_id")
    private Topic topic;

}

File: angularclient/src/app/common/lesson.ts
import {Topic} from "./topic";
import {Time} from "@angular/common";

export class Lesson {

  id: number;
  subject: string;
  date: Date;
  startTime: Time;
  endTime: Time;
  topic: string;

  constructor(subject: string, topic: string, date: Date, startTime: Time, endTime: Time ) {
    this.subject = subject;
    this.topic = topic;
    this.date = date;
    this.startTime = startTime;
    this.endTime = endTime;
  }
}
